File: php/lang/security/audit/sha224-hash.php
<?php

// ruleid: sha224-hash
var_dump(hash('sha224', 'mypassword'));

// ruleid: sha224-hash
var_dump(hash('sha512/224', 'mypassword'));

// ruleid: sha224-hash
var_dump(hash('sha3-224', 'mypassword'));

// ruleid: sha224-hash
var_dump(hash_hmac('sha224', 'mypassword'));

// ruleid: sha224-hash
var_dump(hash_hmac('sha512/224', 'mypassword'));

// ruleid: sha224-hash
var_dump(hash_hmac('sha3-224', 'mypassword'));

// ok: sha224-hash
var_dump(hash('sha384', 'mypassword'));

File: ruby/lang/security/audit/sha224-hash.rb
require 'digest'
class Bad_sha224
    def bad_sha224_code()
        # ruleid: sha224-hash
        sha = Digest::SHA224.hexdigest 'abc'
        # ruleid: sha224-hash
        sha = Digest::SHA224.new
        # ruleid: sha224-hash
        sha = Digest::SHA224.base64digest 'abc'
        # ruleid: sha224-hash
        sha = Digest::SHA224.digest 'abc'

        # ruleid: sha224-hash
        digest = OpenSSL::Digest::SHA224.new
        # ruleid: sha224-hash
        digest = OpenSSL::Digest::SHA224.hexdigest 'abc'
        # ruleid: sha224-hash
        digest = OpenSSL::Digest::SHA224.new
        # ruleid: sha224-hash
        digest = OpenSSL::Digest::SHA224.base64digest 'abc'
        # ruleid: sha224-hash
        digest = OpenSSL::Digest::SHA224.digest 'abc'
        # ruleid: sha224-hash
        OpenSSL::HMAC.hexdigest("sha224", key, data)
        # ok: sha224-hash
        OpenSSL::HMAC.hexdigest("SHA256", key, data)
        # ok: sha224-hash
        digest = OpenSSL::Digest::SHA256.new
        # ok: sha224-hash
        digest = OpenSSL::Digest::SHA256.hexdigest 'abc'

        # ruleid: sha224-hash
        digest = OpenSSL::Digest.new('SHA224')
        
        # ruleid: sha224-hash
        digest = OpenSSL::Digest.new('SHA512-224')

        # ruleid: sha224-hash
        digest = OpenSSL::Digest.new('SHA3-224')

        # ruleid: sha224-hash
        hmac = OpenSSL::HMAC.new(key, 'sha224')

        # ruleid: sha224-hash
        hmac = OpenSSL::HMAC.new(key, 'SHA224')

    end
end
